File: Quadrado Mágico (OBI 2007).c
#include <stdio.h>

#define TRUE 1
#define FALSE 0

int main(void)
{
    int n;
    scanf("%d", &n);
    int array[n][n], i, j;
    
    for (i = 0; i < n; i++)
    {
        for (j = 0; j < n; j++)
            scanf("%d", &array[i][j]);
    }

    int somaL = 0, somaC = 0, diago1 = 0, diago2 = 0, constante = 0, validar = TRUE;


    for (i = 0; i < n && validar == TRUE; i++)
    {
        if (i == 0)
        {
            for ( j = 0; j < n; j++)
                constante += array[i][j];
        }
      
        for (j = 0, somaL = 0, somaC = 0; j < n; j++)
        {
            somaL += array[i][j];
            somaC += array[j][i];
        }
        
        diago1 += array[i][i];
        diago2 += array[i][n-i-1];
        if (i != 0 && (somaL != constante || somaC != constante))
            validar = FALSE;

    }
    
    if (validar && diago1 == constante && diago2 == constante)
    {
        printf("%d\n", constante);
        return 0;
    }
    printf("%d\n", -1);
    return 0;
}

File: other/Quadrado Mágico (OBI 2007).py

n = int(input())
array = list()

for i in range(n):
    array.append([int(iten) for iten in input().split()])

constante, validar = 0, True
somaC = somaL = diag1 = diag2 = 0

for i in range(n):

    if i == 0:
        constante = sum(array[i])

    somaL = sum(array[i])
    somaC = 0
    diag1 += array[i][i]
    diag2 += array[i][n-i-1]

    for coluns in range(n):
        somaC += array[coluns][i]

    if somaL != constante or somaC != constante:
        validar == False
        break

if validar and diag1 == constante and diag2 == constante:
    print(constante)
else:
    print(-1)